src/Components/test/Ignitor/RenderBatchReader.cs

```diff
@@ -0,0 +1,273 @@
+                        var elementName = ReadString(frameData.Slice(8, 4), strings);
+                        result[i / 16] = RenderTreeFrame.Element(0, elementName).WithElementSubtreeLength(elementSubtreeLength);
+                        break;
+
+                    case RenderTreeFrameType.ElementReferenceCapture:
+                        // Client doesn't process these, skip.
+                        result[i / 16] = RenderTreeFrame.ElementReferenceCapture(0, null);
+                        break;
+
+                    case RenderTreeFrameType.Region:
+                        var regionSubtreeLength = BitConverter.ToInt32(frameData.Slice(4, 4));
+                        result[i / 16] = RenderTreeFrame.Region(0).WithRegionSubtreeLength(regionSubtreeLength);
+                        break;
+
+                    case RenderTreeFrameType.Text:
+                        var text = ReadString(frameData.Slice(4, 4), strings);
+                        result[i / 16] = RenderTreeFrame.Text(0, text);
+                        break;
+
+                    case RenderTreeFrameType.Markup:
+                        var markup = ReadString(frameData.Slice(4, 4), strings);
+                        result[i / 16] = RenderTreeFrame.Markup(0, markup);
+                        break;
+
+                    default:
+                        throw new ArgumentException($"Unsupported frame type: {type}");
+                }
+            }
+
+            return new ArrayRange<RenderTreeFrame>(result, result.Length);
+        }
+
+        private static ArrayRange<int> ReadDisposedComponentIds(ReadOnlySpan<byte> data)
+        {
+            return new ArrayRange<int>(Array.Empty<int>(), 0);
+        }
+
+        private static ArrayRange<int> ReadDisposedEventHandlerIds(ReadOnlySpan<byte> data)
+        {
+            return new ArrayRange<int>(Array.Empty<int>(), 0);
+        }
+
+        private static string ReadString(ReadOnlySpan<byte> data, string[] strings)
+        {
+            var index = BitConverter.ToInt32(data.Slice(0, 4));
+            return index >= 0 ? strings[index] : null;
+        }
+
+        private static uint ReadUnsignedLEB128(ReadOnlySpan<byte> data, int startOffset, out int numBytesRead)
+        {
+            var result = (uint)0;
+            var shift = 0;
+            var currentByte = (byte)128;
+            numBytesRead = 0;
+
+            for (var count = 0; count < 4 && currentByte >= 128; count++)
+            {
+                currentByte = data[startOffset + count];
+                result += (uint)(currentByte & 0x7f) << shift;
+                shift += 7;
+                numBytesRead++;
+            }
+
+            return result;
+        }
+
+        private readonly struct Sections
+        {
+            public static Sections Parse(ReadOnlySpan<byte> data)
+            {
+                return new Sections(
+                    BitConverter.ToInt32(data.Slice(data.Length - 20, 4)),
```

Comment:
Like with the `SequenceReader`, this would basically be `new Sections(reader.ReadInt32(), reader.ReadInt32(), ...)`. But I may be overlooking something.